lib/settings.js

```diff
@@ -94,12 +94,30 @@ class Settings extends EventEmitter {
       this._defaultSettingsFilePath;
   }
 
+  _buildWatchPaths(path) {
+    const children = path.split('.');
+    const paths = [];
+
+    for (let i = 0; i < children.length; i++) {
+      const path = _.take(children, i + 1);
+      paths.push(_.join(path, '.'));
+    }
```

Comment:
I think the notification should be more specific to less specific

ex:
a.b.c.d
a.b.c
a.b
a

Reply:
Ugh, instead of reverse, just change the for loop order from children.length to 0?

Also, perhaps rename children to parts.